org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/editor/WRecordIDEditor.java

```diff
@@ -0,0 +1,367 @@
+        {
+			popupMenu = new WEditorPopupMenu(true, true, isShowPreference());
+			popupMenu.addMenuListener(this);
+			addChangeLogMenu(popupMenu);
+        }
+	}
+	
+	public void actionRefresh(Object value)
+    {
+		setValue(value);
+    }
+	
+	@Override
+	public void actionZoom()
+	{
+		Integer tableID = (Integer) tableIDGridField.getValue();
+		Integer recordID = (Integer) gridTab.getValue("Record_ID");
+		if(tableID == null || tableID <= 0 || recordID == null || recordID <= 0)
+			return;
+		if (!MRole.getDefault().isTableAccess(tableID, false))
+			throw new AdempiereException(Msg.getMsg(Env.getCtx(), "AccessTableNoView"));
+		MTable savedTable = new MTable(Env.getCtx(), tableID, null);
+		String[] keyColumns = savedTable.getKeyColumns();
+		String keyColumn = null;
+		if(keyColumns.length > 0) {
+			keyColumn = keyColumns[0];	//	guess
+		}
+	
+		String where = keyColumn + "=" + recordID;
+		PO po = new Query(Env.getCtx(), savedTable, where, null).first();
+		boolean isSOTrx = true;
+		if(po.columnExists("IsSOTrx")) {
+			isSOTrx = po.get_ValueAsBoolean("IsSOTrx");
+		}
+		int AD_Window_ID;
+		if(isSOTrx)
+			AD_Window_ID = savedTable.getAD_Window_ID();
+		else
+			AD_Window_ID = savedTable.getPO_Window_ID();
+
+		if (AD_Window_ID > 0) {
+			if(!Util.isEmpty(keyColumn)) {
+				MQuery zoomQuery = new MQuery();
+				zoomQuery.addRestriction(keyColumn, MQuery.EQUAL, (Integer)po.get_Value(keyColumn));
+				zoomQuery.setRecordCount(1);
+	
+				AEnv.zoom(AD_Window_ID, zoomQuery);
+			}
+		}
+	}
+
+	@Override
+	public void onMenu(ContextMenuEvent evt)
+	{
+		if (WEditorPopupMenu.REQUERY_EVENT.equals(evt.getContextEvent()))
+		{
+			actionRefresh(oldValue);
+		}
+		else if (WEditorPopupMenu.ZOOM_EVENT.equals(evt.getContextEvent()))
+		{
+			actionZoom();
+		}
+		else if (WEditorPopupMenu.CHANGE_LOG_EVENT.equals(evt.getContextEvent()))
+		{
+			WFieldRecordInfo.start(gridField);
+		} 
+		else if (WEditorPopupMenu.PREFERENCE_EVENT.equals(evt.getContextEvent()))
+		{
+			if (isShowPreference())
+				ValuePreference.start (getComponent(), gridField, getValue());
+		}
+	}
+
+	@Override
+	public void setReadWrite(boolean readWrite) {
+		m_ReadWrite = readWrite;
+		if(editButton != null)
+			editButton.setDisabled(!readWrite);
+		
+	}
+
+	@Override
+	public boolean isReadWrite() {
+		return m_ReadWrite;
+	}
+
+	public  void propertyChange(PropertyChangeEvent evt)
+	{
+		if (GridField.PROPERTY.equals(evt.getPropertyName()))
+		{
+			setValue(evt.getNewValue(), false);
+		}
+	}
+	
+	@Override
+	public void setValue(Object value) {
+		setValue(value, true);
+	}
+	
+	/**
+	 * 	Set Value
+	 *	@param value value
+	 *	@param fire data binding
+	 */
+	
+	private void setValue (Object value, boolean fire) {
+		if (value == null || value.toString().trim().length() == 0)
+		{
+			oldValue = null;
+			value = null;
+			recordTextBox.setValue("");
+		}
+		else {
+			if(oldTableID != tableIDGridField.getValue()) {
+				oldTableID = tableIDGridField.getValue();
+			}
```

Comment:
should just update oldTableID from WRecordIDDialog, i.e call editor.setAD_Table_ID(..) in WRecordIDDialog